### PR TITLE
Update to numba version 0.42.1

### PR DESCRIPTION
**Pull request by pearu**
Checklist
* [x] Used a fork of the feedstock to propose changes
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/conda_smithy.html#how-to-re-render ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

Restricting to numpy version 1.15.x as all tests pass ok. With numpy 1.16.1 there are 77 failures and 3 errors in numba tests.

Replaces PR #21 and PR #22

### CODE REVIEW

**Comment by pearu** [comment]
@conda-forge-admin, please rerender

**Comment by pearu** [comment]
@souravsingh @marcelotrevisani @jakirkham, please review.

**Comment by pearu** [comment]
2nd try: @souravsingh @marcelotrevisani @jakirkham, please review.

**Comment by souravsingh** [comment]
LGTM

File: recipe/meta.yaml [hunk 1]
```diff
@@ -42,6 +44,8 @@ requirements:
     - python
     - {{ pin_compatible('llvmlite', max_pin='x.x') }}
     - {{ pin_compatible('numpy') }}
+    - llvmlite 0.27.* # otherwise conda test picks up llvmlite >= 0.28
+    - numpy 1.15.*
```
Comment (jakirkham) [inline recipe/meta.yaml]: Am curious why this line was needed (given we have `{{ pin_compatible('numpy') }}` above)? Does anyone remember? Also would it be possible to drop it?

Comment (pearu) [reply recipe/meta.yaml]: numpy 1.16 introduced NaT that makes time-related numba tests fail. I am not sure if the numba master is NaT aware.

Comment (pearu) [reply recipe/meta.yaml]: It looks like numba 0.43.1 has numpy 1.16 NaT support implemented.

**Pull request merged**